### PR TITLE
Add C api for RateLimiter

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Add C api for RateLimiter.

### CODE REVIEW

**Comment by siddontang** [comment]
PTAL @siying

**Comment by yiwu-arbug** [comment]
@zhangjinpeng1987 Can you sign the CLA? Thanks!

File: db/c_test.c [hunk 1]
```diff
@@ -326,6 +327,9 @@ int main(int argc, char** argv) {
   int compression_levels[] = {rocksdb_no_compression, rocksdb_no_compression,
                               rocksdb_no_compression, rocksdb_no_compression};
   rocksdb_options_set_compression_per_level(options, compression_levels, 4);
+  rate_limiter = rocksdb_ratelimiter_create(100*1024*1024, 100*1000, 10);
+  rocksdb_options_set_ratelimiter(options, rate_limiter);
+  rocksdb_ratelimiter_destroy(rate_limiter);
```
Comment (yiwu-arbug) [inline db/c_test.c]: You didn't declare `rocksdb_ratelimiter_destroy` in c.h.

**Comment by zhangjinpeng1987** [comment]
Seems `DBTest_FileCreationRandomFailure_Test` will failed randomly.

**Pull request closed without merging**